File: app/src/main/java/com/aplicacion/mypet/providers/TokenProvider.java
package com.aplicacion.mypet.providers;

import com.aplicacion.mypet.models.Token;
import com.google.android.gms.tasks.OnSuccessListener;
import com.google.android.gms.tasks.Task;
import com.google.firebase.firestore.CollectionReference;
import com.google.firebase.firestore.DocumentSnapshot;
import com.google.firebase.firestore.FirebaseFirestore;
import com.google.firebase.messaging.FirebaseMessaging;

public class TokenProvider {
    private CollectionReference collectionReference;

    public  TokenProvider() {
        collectionReference = FirebaseFirestore.getInstance().collection("Tokens");
    }

    public void create(final String idUser) {
        if (idUser==null) {
            return;
        }

        FirebaseMessaging.getInstance().getToken().addOnSuccessListener(new OnSuccessListener<String>() {
            @Override
            public void onSuccess(String s) {
                Token token = new Token(s);
                collectionReference.document(idUser).set(token);
            }
        });
    }

    public Task<DocumentSnapshot> getToken(String idUser) {
        return collectionReference.document(idUser).get();
    }
}

File: app/src/main/java/com/aplicacion/mypet/providers/TokenProvider.kt
package com.aplicacion.mypet.providers

import com.aplicacion.mypet.models.Token
import com.google.android.gms.tasks.Task
import com.google.firebase.firestore.CollectionReference
import com.google.firebase.firestore.DocumentSnapshot
import com.google.firebase.firestore.FirebaseFirestore
import com.google.firebase.messaging.FirebaseMessaging
class TokenProvider() {
    private lateinit var collectionReference: CollectionReference

    init {
        collectionReference = FirebaseFirestore.getInstance().collection("Tokens")
    }

    fun create(idUser: String?) {
        if (idUser == null) {
            return
        }

        FirebaseMessaging.getInstance().token.addOnSuccessListener { s ->
            val token = Token(s)
            collectionReference.document(idUser).set(token)
        }
    }

    fun getToken(idUser: String?): Task<DocumentSnapshot?> {
        return collectionReference.document(idUser!!).get()
    }

}